pandas/core/groupby/generic.py

```diff
@@ -1308,42 +1308,35 @@ def _transform_general(self, func, *args, **kwargs):
         gen = self.grouper.get_iterator(obj, axis=self.axis)
         fast_path, slow_path = self._define_paths(func, *args, **kwargs)
 
-        for name, group in gen:
-            if group.size == 0:
-                continue
+        # Determine whether to use slow or fast path by evaluating on the first group.
+        # Need to handle the case of an empty generator and process the result so that
+        # it does not need to be computed again.
+        try:
+            name, group = next(gen)
```

Comment:
do we need to keep going if the first group is size 0?

Reply:
Yes, for categorical with `obsevered=False`